From 3a8c41604de0829a5ca7eeb14d8df9142ea47e27 Mon Sep 17 00:00:00 2001
From: Mathew McBride <matt@traverse.com.au>
Date: Thu, 28 Aug 2025 13:18:42 +1000
Subject: [PATCH 12/12] firewall: only attempt to execute MAC filter script if
 it exists

If the MAC filter script (/tmp/mac_filter.sh) has not been generated,
this can result in excessive output to the system console, such
as this:

/bin/sh: /tmp/mac_filter.sh: No such file or directory

Change-Id: Ia7556bbed9a78a6b6d983da94f0e3842b2cd3ded
Signed-off-by: Mathew McBride <matt@traverse.com.au>
---
 source/firewall/firewall.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source/firewall/firewall.c b/source/firewall/firewall.c
index b7093c29..0a91013f 100644
--- a/source/firewall/firewall.c
+++ b/source/firewall/firewall.c
@@ -16814,6 +16814,7 @@ static int service_start ()
    /* Applies to all of Raspberry Pi, BPI-R4 and Generic Arm */
    #ifdef _GENERIC_LINUX_DATA_PATH
        /* Apply Mac Filtering rules for RPI-Device */
+   if (access("/tmp/mac_filter.sh", F_OK) == 0)
        v_secure_system("/bin/sh -c /tmp/mac_filter.sh");
    #endif
 
-- 
2.39.5

